core/src/main/scala/com/salesforce/op/OpWorkflow.scala

```diff
@@ -234,10 +234,13 @@ class OpWorkflow(val uid: String = UID[OpWorkflow]) extends OpWorkflowCore {
               "The RawFeatureFilter training reader will be used to generate the data for training")
         }
         checkReadersAndFeatures()
-        val FilteredRawData(cleanedData, featuresToDrop, mapKeysToDrop, featureDistributions) =
+
+        val FilteredRawData(cleanedData, featuresToDrop, mapKeysToDrop, rawFeatureFilterResults) =
           rf.generateFilteredRaw(rawFeatures, parameters)
-        setRawFeatureDistributions(featureDistributions.toArray)
-        setBlacklist(featuresToDrop, featureDistributions)
+
+        setRawFeatureFilterResults(rawFeatureFilterResults)
+        setRawFeatureDistributions(rawFeatureFilterResults.featureDistributions.toArray)
```

Comment:
can you delete the rawFeatureDistributions variable and have the getter methods point to rawFeatureFilterResults?

Reply:
You will also need to modify the workflow reader and writer so this information is stored across loaded workflows
https://github.com/salesforce/TransmogrifAI/blob/master/core/src/main/scala/com/salesforce/op/OpWorkflowModelWriter.scala#L84
https://github.com/salesforce/TransmogrifAI/blob/master/core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala#L96

Reply:
Good point. In the reader we can make it backwards compatible same as we did for black list of features